diesel/src/backend.rs

```diff
@@ -264,9 +265,23 @@ pub mod sql_dialect {
         /// `QueryFragment` implementations
         pub trait SupportsOnConflictClause {}
 
+        /// Indicates that a backend provides support for `ON CONFLICT` clauses
+        /// using the postgresql `ON CONFLICT` syntax
+        #[derive(Debug, Copy, Clone)]
+        pub struct PgLikeOnConflictClause;
+
+        impl SupportsOnConflictClause for PgLikeOnConflictClause {}
+
```

Comment:
As this and the `MysqlLikeOnConflictClause` type is backend specific they should be part of the corresponding backend module. If any `QueryFragment` implementation refers to those types, that's an indication that this impl should moved to `backend/query_builder/…` as well.

Reply:
Yeah I'm definitely still a bit confused on the matter, but I was following [the code](https://github.com/diesel-rs/diesel/blob/42413462fbb67d6b4b612f1f62ba9f8af465d920/diesel/src/query_builder/returning_clause.rs#L30-L54) you had previously pointed me to as an example to work off (though I am not sure if it's the best example as I'm still kind of in just-get-it-to-compile mode 😅 , rather than fully understanding how such generic code is organized)

Where
`ReturningClause = crate::backend::sql_dialect::returning_clause::PgLikeReturningClause`

So what I was trying to do is basically something along the lines of
`OnConflictClause = ...::PgLikeOnConflictClause`